### PR TITLE
Create `SitterProfileView` and integrate it to the `SitterMainTabView`

### LINKED ISSUE #114: Create pet sitter profile view
## This task is a part of:
 - #9 
 

## Depends on tasks:
- #61
- #112
- #113 

## Description
To tap on the profile button user will be able to see info about themselves and the edit button.

When this screen appears the first time name is displayed like "User" and other fields will be empty.

When the sitter switches to edit mode by tapping `Edit` button on the `SitterProfileView`, the 2 options should be provided:
    - `Cancel` -  to close the edit view without any changes.
    - `Save` - to save changes.

If the name textfield will be empty the save button will be disabled and the prompt "name is a mandatory field" will be shown.

### Dev requirements and restrictions:

- Create a view that contains:
   - SitterCardView + Edit button
   - SitterInfoEditView + 2 buttons (Save and Cancel).
- Integrate view into SitterMainTabView instead of stab.

## UI Mocks
A simplified UI design is below, please follow the [App Style Guidelines](https://github.com/ios-course/ironfoudation-team-project/wiki/App-style-guidelines)
Screenshot 2023-06-12 at 22 34 08
Screenshot 2023-06-12 at 22 33 49

### PR DESCRIPTION
**Pull request by YuliaNastiushkina**
## Summary
This PR is a part of the work to resolve #114.

## UI Preview (Screenshots)
<details> 
<Summary>Tap to show screenshots  </Summary>

<details>

<Summary>iPhone 14 Pro -  6.1"</Summary>
Screenshot 2023-06-21 at 11 58 06

</details>
<details>

<Summary>iPhone 14 Pro Max - 6.7"</Summary>
Screenshot 2023-06-21 at 11 57 49

</details>
<details>

<Summary>iPhone SE 3rd gen - 4.7"</Summary>
Screenshot 2023-06-21 at 11 58 47

</details>
</details>

## Changes
- added view `SitterProfileView`
- replace type `Double` to `String` for `ratePerHour` in `SitterInfoCardView`
- integrate view into `SitterMainTabView`

## Checks to complete
- [x] I've built and run my changes, and no warnings or errors occurred.
- [x] All existing tests passed with my changes.
- [x] My code follows our style guide.
- [x] I've performed a self-review of my code.
- [x] PR's title has a conventional commit style (short and descriptive).
- [x] Every non-private interface is documented properly.
- [x] I've added tests for my code (if it makes sense).
- [x] PR has assignee, reviewers, milestone, it is properly linked to the project and to the issue.

### CODE REVIEW

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterInfoCardView.swift [hunk 1]
```diff
@@ -74,38 +74,21 @@ struct SitterInfoCardView: View {
     private let lineLimitForSurname = 1
     private let linelimitForPhoneNumber = 1
     private let linelimitForBio = 10
+    private let defaultCurrency = "$"
 }
 
 struct SitterInfoCardView_Previews: PreviewProvider {
     static var previews: some View {
+        let viewModel = SitterProfileViewModel()
```
Comment (kirpichonok) [inline Woof/Woof/View/SitterFlow/SitterProfileView/SitterInfoCardView.swift:82]: Should we integrate the view model into the view? Why do you use this view model separate from the view?

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                        surname: viewModel.surname,
+                        phone: viewModel.phone,
+                        bio: viewModel.bio,
+                        ratePerHour: viewModel.pricePerHour
+                    )
+
+                    Button("Edit") {
+                        isEditingMode.toggle()
+                    }
+                    .disabled(viewModel.name.isEmpty)
```
Comment (marinkman) [inline Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift:51]: Should this button become inactive if there is no name?

Comment (marinkman) [inline Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift:51]: Should this button become inactive if there is no name?

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+                    HStack {
+                        Button("Save") {
+                            viewModel.save()
+                            isEditingMode.toggle()
+                        }
+
+                        Spacer()
+
+                        Button("Cancel") {
```
Comment (marinkman) [inline Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift:29]: When I click the "edit" button, change some data and click the "cancel" button, I expect that data will not be displayed in my profile. 
And this data is displayed. 
This confuses me.

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift (hunk 2 above)
Comment (marinkman) [inline Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift:51]: For now, I can save changes with no name, but then I won't be able to change my profile information at all.

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                        phone: $viewModel.phone,
+                        bio: $viewModel.bio,
+                        pricePerHour: $viewModel.pricePerHour
+                    )
+
+                    HStack {
+                        Button("Save") {
+                            viewModel.save()
+                            isEditingMode.toggle()
+                        }
```
Comment (marinkman) [inline Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift:25]: The `save` button should be disabled when there is no name.

**Review by marinkman — CHANGES_REQUESTED** [review]
Good start, but I have a few remarks.
Please fix this:

**Review by marinkman — COMMENTED** [review]
Also you forgot to make changes to the `CHANGELOG.md` file.

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift [hunk 5]
```diff
@@ -0,0 +1,82 @@
+import SwiftUI
+
```
Comment (kirpichonok) [inline Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift:2]: Missed documentation.

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+                    EditSitterInformationView(
+                        name: $viewModel.name,
+                        surname: $viewModel.surname,
+                        phone: $viewModel.phone,
+                        bio: $viewModel.bio,
+                        pricePerHour: $viewModel.pricePerHour
+                    )
+
+                    HStack {
+                        Button("Save") {
```
Comment (kirpichonok) [inline Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift:22]: Magic strings. See the rest of code.

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+                        Button("Cancel") {
+                            isEditingMode.toggle()
+                        }
+                    }
+                }
+                .padding()
+                .buttonStyle(CapsuleWithWhiteText())
+                .background(Color.App.purpleLight)
+                .cornerRadius(AppStyle.UIElementConstant.cornerRadius)
+            } else {
```
Comment (kirpichonok) [inline Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift:38]:
```suggestion

            } else {
```
 Please, add an empty line to improve readability.

**Review by kirpichonok — CHANGES_REQUESTED** [review]
Good attempt! 
But the problem that when we tap button `Cancel` on the `SitterProfileView` the entered changes are saved.
And don't remember to update changelog.

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift (hunk 2 above)
Comment (YuliaNastiushkina) [reply Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift]: No, it doesn't. Fixed it, thanks!

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift (hunk 3 above)
Comment (YuliaNastiushkina) [reply Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift]: Yep, my fault, I fixed it. Check how it works now.

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift (hunk 2 above)
Comment (YuliaNastiushkina) [reply Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift]: It shouldn't work like that, you're right, it my mistake. I fixed it.

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift (hunk 4 above)
Comment (YuliaNastiushkina) [reply Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift]: Yes, fixed it.

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift (hunk 5 above)
Comment (YuliaNastiushkina) [reply Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift:2]: Fixed.

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift (hunk 6 above)
Comment (YuliaNastiushkina) [reply Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift]: Yes, sorry, fixed it.

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift (hunk 7 above)
Comment (YuliaNastiushkina) [reply Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift:45]: Fixed.

**Review by NadzeyaShpakouskaya — CHANGES_REQUESTED** [review]
Consider minor improvements

File: CHANGELOG.md [hunk 8]
```diff
@@ -31,6 +31,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 - Create `PurpleCapsuleOfInfinityWidth` as `ButtonStyle`. Create `LoginView` and apply in the `WoofApp` to display when running the app when the user role is not selected - [#160](https://github.com/ios-course/ironfoudation-team-project/pull/160)
 - Add the `EditSitterInformationView` and the reusable element `TextEditorWithPlaceholder` - [#159](https://github.com/ios-course/ironfoudation-team-project/pull/159)
 - Add ability to change the user role using `Logout` button on the `SitterMainTabView` and `OwnerMainTabView` screens - [#167](https://github.com/ios-course/ironfoudation-team-project/pull/167)
+- Add `SitterProfileView` to display and edit personal information about the sitter - [#114](https://github.com/ios-course/ironfoudation-team-project/issues/114)
```
Comment (marinkman) [inline CHANGELOG.md:34]: It looks like you made the changes in the wrong place.
Please move these changes to the `Unreleased` section.

Comment (YuliaNastiushkina) [reply CHANGELOG.md]: Yes, thanks, fixed it.

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+        viewModel.pricePerHour = viewModelCopy.pricePerHour
+    }
+
+    private func updateValues() {
+        viewModelCopy.name = viewModel.name
+        viewModelCopy.surname = viewModel.surname
+        viewModelCopy.phone = viewModel.phone
+        viewModelCopy.bio = viewModel.bio
+        viewModelCopy.pricePerHour = viewModel.pricePerHour
+    }
```
[the remark this answers is not in the document]
Comment (YuliaNastiushkina) [reply Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift]: Yes, it looks much better, thanks!
Fixed it.

File: CHANGELOG.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (YuliaNastiushkina) [reply CHANGELOG.md]: Sorry, fixed it.

**Review by julyakmita — APPROVED** [review]
Great improvement. 
LGTM.

**Review by NadzeyaShpakouskaya — APPROVED** [review]
Overall LGTM, just move your changes in CHANGELOG file to `Added` section instead of `Changed`

File: Woof/Woof/ViewModel/SitterProfileViewModel.swift [hunk 10]
```diff
@@ -52,7 +52,7 @@ final class SitterProfileViewModel: ObservableObject {
 
     // MARK: - Private interface
 
-    private lazy var currentSitter: Sitter = loadSitterFromStorage()
+    private(set) lazy var currentSitter: Sitter = loadSitterFromStorage()
```
Comment (kirpichonok) [inline Woof/Woof/ViewModel/SitterProfileViewModel.swift:55]:
```suggestion
    private lazy var currentSitter: Sitter = loadSitterFromStorage()
```
Why do we need `set` here?

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift [hunk 11]
```diff
@@ -45,24 +51,35 @@ struct SitterProfileView: View {
...
+    private let saveButtonLabelText = "Save"
+    private let editButtonLabelText = "Edit"
     @State private var isEditingMode = false
+
+    private func cancelEditing() {
+        viewModel.name = viewModel.currentSitter.name
+        viewModel.surname = viewModel.currentSitter.surname
+        viewModel.phone = viewModel.currentSitter.phone
+        viewModel.bio = viewModel.currentSitter.bio
+        viewModel.pricePerHour = viewModel.currentSitter.pricePerHour.description
```
Comment (kirpichonok) [inline Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift:81]: I think you should move this logic to the view model.

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift [hunk 12]
```diff
@@ -19,22 +20,27 @@ struct SitterProfileView: View {
                     )
 
                     HStack {
-                        Button("Save") {
-                            viewModel.save()
-                            isEditingMode.toggle()
-                        }
+                        if isEditingMode {
```
Comment (kirpichonok) [inline Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift:23]: Why do you need this checking here?

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift [hunk 13]
```diff
@@ -19,22 +20,27 @@ struct SitterProfileView: View {
...
+                            Button(cancelButtonLabelText) {
+                                cancelEditing()
```
Comment (kirpichonok) [inline Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift:25]:
```suggestion
                                viewModel.cancelEditing()
```
it should look something like this.

**Review by kirpichonok — CHANGES_REQUESTED** [review]
Almost good, consider improvements.

File: Woof/Woof/ViewModel/SitterProfileViewModel.swift (hunk 10 above)
Comment (YuliaNastiushkina) [reply Woof/Woof/ViewModel/SitterProfileViewModel.swift]: Now we don't need it.
Removed.

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift (hunk 11 above)
Comment (YuliaNastiushkina) [reply Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift]: You're right, thanks.
Fixed it.

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift (hunk 12 above)
Comment (YuliaNastiushkina) [reply Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift]: This check was superfluous, deleted it.

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift (hunk 13 above)
Comment (YuliaNastiushkina) [reply Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift]: Fixed.

File: CHANGELOG.md [hunk 14]
```diff
@@ -15,6 +15,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 ### Changed
 
  - Update PR template [#194](https://github.com/ios-course/ironfoudation-team-project/pull/194)
+- Add `SitterProfileView` to display and edit personal information about the sitter - [#190](https://github.com/ios-course/ironfoudation-team-project/pull/190)
```
Comment (marinkman) [inline CHANGELOG.md:18]: Please move your changes to `Added` section.

Comment (YuliaNastiushkina) [reply CHANGELOG.md]: Sorry, didn't check after resolving the merge conflict.
Fixed.

**Review by marinkman — APPROVED** [review]
LGTM

File: Woof/Woof/ViewModel/SitterProfileViewModel.swift [hunk 15]
```diff
@@ -65,4 +70,12 @@ final class SitterProfileViewModel: ObservableObject {
 
         return sitter
     }
+
+    private func restoreOriginalValues(_ currentSitter: Sitter) {
```
Comment (kirpichonok) [inline Woof/Woof/ViewModel/SitterProfileViewModel.swift:74]: Can we use this function in the init? DRY

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelCancelEditingTests.swift [hunk 16]
```diff
@@ -0,0 +1,36 @@
+import XCTest
+
+final class SitterProfileViewModelCancelEditingTests: XCTestCase {
+    func testCancelEditingMethodExistsInAPI() {
+        SitterProfileViewModel().cancelEditing()
+    }
+
+    func testCancelEditingMethodSuccessfullyRestoresOriginalSitterProperties() {
```
Comment (kirpichonok) [inline Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelCancelEditingTests.swift:8]: I didn't understand your intentions in this test. What do you test? Can you explain?

**Review by kirpichonok — CHANGES_REQUESTED** [review]
It looks better, but I have a few questions.

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelCancelEditingTests.swift [hunk 17]
```diff
@@ -0,0 +1,36 @@
+import XCTest
+
+final class SitterProfileViewModelCancelEditingTests: XCTestCase {
+    func testCancelEditingMethodExistsInAPI() {
```
Comment (kirpichonok) [inline Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelCancelEditingTests.swift:4]:
```suggestion
    func testExistsInAPI() {
```
nit: You have not to write the name of the method in functions, if the name of the test class unambiguously indicates a method under tests.

File: Woof/Woof/ViewModel/SitterProfileViewModel.swift (hunk 15 above)
Comment (YuliaNastiushkina) [reply Woof/Woof/ViewModel/SitterProfileViewModel.swift]: Fixed.

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelCancelEditingTests.swift (hunk 17 above)
Comment (YuliaNastiushkina) [reply Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelCancelEditingTests.swift:4]: I followed the general naming style that is used in the project. But, I'll keep that in mind for the future, thanks.

File: Woof/Woof/ViewModel/SitterProfileViewModel.swift [hunk 18]
```diff
@@ -27,11 +27,7 @@ final class SitterProfileViewModel: ObservableObject {
     init() {
         currentSitter = loadSitterFromStorage()
 
-        name = currentSitter.name
-        surname = currentSitter.surname
-        phone = currentSitter.phone
-        bio = currentSitter.bio
-        pricePerHour = String(currentSitter.pricePerHour)
+        storeInitialValues(currentSitter)
```
Comment (kirpichonok) [inline Woof/Woof/ViewModel/SitterProfileViewModel.swift:30]:
```suggestion
        setFieldsAccordingSitter(currentSitter)
```
What do you think about it?

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelCancelEditingTests.swift [hunk 19]
```diff
@@ -9,28 +9,32 @@ final class SitterProfileViewModelCancelEditingTests: XCTestCase {
...
+        let originalPhone = viewModel.phone
+        let originalBio = viewModel.bio
+        let originalPricePerHour = viewModel.pricePerHour
 
-        viewModel.name = newName
-        viewModel.surname = newSurname
-        viewModel.phone = newPhone
-        viewModel.bio = newBio
-        viewModel.pricePerHour = String(newPricePerHour)
+        viewModel.name = Sitter.Test.johnSmith.name
```
Comment (kirpichonok) [inline Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelCancelEditingTests.swift:18]:
```suggestion
       // When
        viewModel.name = Sitter.Test.johnSmith.name
```
Would it be better to place it here?

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelCancelEditingTests.swift [hunk 20]
```diff
@@ -9,28 +9,32 @@ final class SitterProfileViewModelCancelEditingTests: XCTestCase {
...
-        XCTAssertEqual(viewModel.name, testSitter.name)
-        XCTAssertEqual(viewModel.surname, testSitter.surname)
-        XCTAssertEqual(viewModel.phone, testSitter.phone)
-        XCTAssertEqual(viewModel.bio, testSitter.bio)
-        XCTAssertEqual(viewModel.pricePerHour, String(testSitter.pricePerHour))
+        XCTAssertNotEqual(viewModel.name, Sitter.Test.johnSmith.name)
+        XCTAssertNotEqual(viewModel.surname, Sitter.Test.johnSmith.surname)
+        XCTAssertNotEqual(viewModel.phone, Sitter.Test.johnSmith.phone)
+        XCTAssertNotEqual(viewModel.bio, Sitter.Test.johnSmith.bio)
+        XCTAssertNotEqual(viewModel.pricePerHour, String(Sitter.Test.johnSmith.pricePerHour))
```
Comment (kirpichonok) [inline Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelCancelEditingTests.swift:38]: We have checked the equality on in the asserts above.  Does this checking make sense?

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelCancelEditingTests.swift (hunk 19 above)
Comment (YuliaNastiushkina) [reply Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelCancelEditingTests.swift:18]: Discussed it. Will stay as it is.

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelCancelEditingTests.swift (hunk 20 above)
Comment (YuliaNastiushkina) [reply Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelCancelEditingTests.swift:38]: This is extra checking to ensure that the current value of fields doesn't equal values that were set before cancel editing.

**Review by kirpichonok — APPROVED** [review]
LGTM

**Pull request merged**